companion/src/firmwares/radiodata.cpp

```diff
@@ -106,76 +106,75 @@ void RadioData::convert(RadioDataConversionState & cstate)
 
 void RadioData::addLabel(QString label)
 {
-  label.replace("/c",",");
-  label.replace("//","/");
-  if(labels.indexOf(label) == -1)
+  label = unEscapeCSV(label);
+  if (labels.indexOf(label) == -1)
     labels.append(label);
 }
 
 bool RadioData::deleteLabel(QString label)
 {
   bool deleted = false;
-  QString csvLabel = label;
-  csvLabel.replace("/","//");
-  csvLabel.replace(",","/c");
 
   // Remove labels in the models
   for(auto& model : models) {
-    QStringList modelLabels = QString(model.labels).split(',',QString::SkipEmptyParts);
-    if(modelLabels.indexOf(csvLabel) >= 0)
+    QStringList modelLabels = fromCSV(model.labels);
+    if (modelLabels.indexOf(label) >= 0) {
       deleted = true;
-    modelLabels.removeAll(csvLabel);
-    strcpy(model.labels, QString(modelLabels.join(',')).toLocal8Bit().data());
+      modelLabels.removeAll(label);
+    }
+    strcpy(model.labels, toCSV(modelLabels).toLocal8Bit().data());
   }
 
   // Remove the label from the global list
   labels.removeAll(label);
 
   // If no labels remain, add a Favorites one
-  if(!labels.size()) {
+  if (!labels.size()) {
     addLabel(tr("Favorites"));
   }
   return deleted;
 }
 
 bool RadioData::deleteLabel(int index)
 {
-  if(index >= labels.size()) return false;
+  if (index >= labels.size()) return false;
   QString modelLabel = labels.at(index);
   return deleteLabel(modelLabel);
 }
 
 bool RadioData::renameLabel(QString from, QString to)
 {
-  bool success = true;
-  int lengthdiff = to.size() - from.size();
-  QString csvFrom = from;
-  QString csvTo = to;
-  csvFrom.replace("/","//");
-  csvFrom.replace(",","/c");
-  csvTo.replace("/","//");
-  csvTo.replace(",","/c");
-
+  bool success = true;  
+  QString csvFrom = escapeCSV(from);
+  QString csvTo = escapeCSV(to);
+  int lengthdiff = csvTo.size() - csvFrom.size();
 
-  // Check that rename is possible, not too long
+  // Check that rename is possible (Rename won't cause too long of a string)
   for(auto& model : models) {
-    if((int)strlen(model.labels) + lengthdiff > (int)sizeof(model.labels) - 1) {
-      success = false;
+    if ((int)strlen(model.labels) + lengthdiff > (int)sizeof(model.labels) - 1) {
+      success = false;     
+      throw std::length_error(model.name);
```

Comment:
@elecpower Since radio data isn't a Q_OBJECT I can't emit a signal for a message. Are you okay with this, or should I re-work to make RadioData a Q_OBJECT and inherit it. Little more of a challenge with the use of RadioData copy constructor

Or something else...

Reply:
Yes it should be a Q_OBJECT and this discussion came up in OTX years ago but was dropped for fear of what it would break and the time to test and fix.
I try to create all new classes as QObjects or QWidgets to take advantage of Qt framework even if they do not need if the first iteration.
You are effectively doing this with no break risk so I'm okay with your approach in the instance